### ISSUE TITLE
Design of replacing flash.sh script with Python script reusing OSFV lib

### ISSUE THREAD

**Issue by macpijan** [3mdeb/meta-rte#82]
We do already have some Python module: https://github.com/Dasharo/osfv-scripts/blob/main/osfv_cli/src/osfv/libs/rte.py
which already implements what we need in this script: https://github.com/3mdeb/RteCtrl/pull/22/files

We should review both of them, and propose if/how we can achieve code reuse, by implementing local `flash.py` script on RTE, using the same module, simply using localhost address as `rte_ip`.

This script may reside in `osfv-scripts` repo.

**Comment by macpijan** [comment]
Maybe there is no need for another script, and we could somehow use `osfv_cli` directly on the localhost.

I think resolving this https://github.com/Dasharo/osfv-scripts/issues/57 and possible also adding one or two more features, could help to achieve this.

We need to first take a look at possible solutions and decide on something.

**Comment by DaniilKl** [comment]
It could be done by installing `osfv_cli` on RTE and using `localhost` as RTE IP. But `osfv_cli` as well as `RteCtrl` will need to be extended for WP features and for using custom `flashrom` parameters in the way `flash.sh` does. Currently it is not implemented. We can also consider using SSH or REST API to do the work, though IMO REST API is more suitable to use with `localhost`, but in that case, `osfv_cli` will need to be modified to use `flashrom` via REST API, because currently it is using SSH.

And, most importantly, the issue https://github.com/Dasharo/osfv-scripts/issues/57, must be solved, otherwise this issue could not be solved.

Plan:
1. - [ ] Solve https://github.com/Dasharo/osfv-scripts/issues/57.
2. - [ ] Make `osfv_cli` use `flashrom` via HTTP instead of SSH for localhost (optional). 
3. - [ ] Add providing additional parameters to `flashrom`
4. - [ ] Add WP options according to `flash.sh` script

**Comment by macpijan** [comment]
> Make osfv_cli use flashrom via HTTP instead of SSH for localhost (optional).

Could be done, but what would be the main benefit? I would also avoid switching to HTTP **just** for localhost. It would need to be done in general. BTW, we already have some of it implemented: https://github.com/3mdeb/RteCtrl/tree/master/pkg/flashromControl

> Add providing additional parameters to flashrom

Correct, at minimum some additional flag to `flash` commmand would do.

>  Add WP options according to flash.sh script

This can in fact be added, but no currently active platforms needs it AFAIK. We do not use this option of this script. Low priority.

**Comment by DaniilKl** [comment]
> Could be done, but what would be the main benefit?

I do not see any disadvantages of the SSH solution, therefore marked it as `optional`. We could skip it if you think it is unnecessary.

**Comment by WiktorG351** [comment]
@macpijan does this approach seem right?
https://github.com/Dasharo/osfv-scripts/pull/97

**Comment by macpijan** [comment]
No, it does not. You have copied functions around instead of implementing what @DaniilKl has proposed.

**Comment by DaniilKl** [comment]
@WiktorG351, this is how RTE and osfv_cli works with each other:

![Image](https://github.com/user-attachments/assets/4f9fbb90-39f4-46f0-a619-b7d2f6bb0e0e)

This is what was supposed to be done:

![Image](https://github.com/user-attachments/assets/a52a2da6-aa78-45ff-bd9d-d0f754e96237)

This is what you have done:

![Image](https://github.com/user-attachments/assets/a6253663-c5e9-4731-bec1-69d33825bc43)

**Comment by DaniilKl** [comment]
My bad. Replace `0.0.0.0` with `127.0.0.1`.

**Comment by WiktorG351** [comment]
@DaniilKl thanks for the heads up. So to do this we will need to have `osfv_cli` on the RTE - I have started work on a recipe for `osfv_cli` in meta-rte in [this PR](https://github.com/3mdeb/meta-rte/pull/98) - although its not working just yet.
